flang/include/flang/Runtime/descriptor.h

```diff
@@ -420,6 +420,27 @@ class Descriptor {
 
   void Dump(FILE * = stdout) const;
 
+// Value of the addendum presence flag.
+#define _CFI_ADDENDUM_FLAG 1
+// Number of bits needed to be shifted when manipulating the allocator index.
+#define _CFI_ALLOCATOR_IDX_SHIFT 1
+// Allocator index mask.
+#define _CFI_ALLOCATOR_IDX_MASK 0b00001110
+
+  RT_API_ATTRS inline bool HasAddendum() const {
+    return raw_.extra & _CFI_ADDENDUM_FLAG;
+  }
+  RT_API_ATTRS inline void SetHasAddendum() {
+    raw_.extra |= _CFI_ADDENDUM_FLAG;
+  }
+  RT_API_ATTRS inline int GetAllocIdx() const {
+    return (raw_.extra & _CFI_ALLOCATOR_IDX_MASK) >> _CFI_ALLOCATOR_IDX_SHIFT;
+  }
+  RT_API_ATTRS inline void SetAllocIdx(int pos) {
+    raw_.extra &= ~_CFI_ALLOCATOR_IDX_MASK; // Clear the allocator index bits.
+    raw_.extra |= (pos << _CFI_ALLOCATOR_IDX_SHIFT);
```

Comment:
This causes the build to fail, as the following:
```
/mnt/hdd1tb/JENKINS/workspace/update_upstream-main/llvm-project/flang/include/flang/Runtime/descriptor.h:441:16: error: array subscript 'Fortran::runtime::Descriptor[0]' is partly outside array bounds of 'Fortran::runtime::StaticDescriptor<0> [1]' [-Werror=array-bounds]
   441 |     raw_.extra |= (pos << _CFI_ALLOCATOR_IDX_SHIFT);
       |     ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Can you please check?

Reply:
Which compiler are you using? All build bots are working fine with this change.

Reply:
With:
```
	[...]
    -DCMAKE_C_COMPILER=gcc \
    -DCMAKE_CXX_COMPILER=g++ \
    -DFLANG_ENABLE_WERROR=ON
```
I get the error.
With `-DFLANG_ENABLE_WERROR=OFF` gives some WARNING and it works

Reply:
It's hard to have `-DFLANG_ENABLE_WERROR=ON` for any compilers. Usually building with clang is warning free because it is what is used in upstream buildbots. 

I'm often building with gcc 9.3.0 and there is tons of warnings.

Reply:
Yes, got it. Thank you!

Reply:
> I'm often building with gcc 9.3.0 and there is tons of warnings.

I use gcc 9.3.0 with -Werror enabled for flang.  What warnings are you seeing?

Reply:
It doesn't throw the error for gcc 9. I will use gcc 9 only. Thank you!